### PR TITLE
fix(policyMatcher): handling relative path resource by joining it with cwd

### LINKED ISSUE #750: policy matching with relative path access has issues with telemetry/alerts
## Bug Report

KubeArmor requires that you specify an absolute path in the policy rules. However, when the actual access happens with the relative path, the action is honored, but the policy name/severity/message is not populated with appropriate policy details (`defaultPosture` is specified as the policy name).

### Environment description (GKE, VM-Kubeadm, vagrant-dev-env, minikube, microk8s, ...)
I tried with k3s

### Kernel version (run `uname -a`)
`Linux ubuntu2004-vagrants 5.4.0-120-generic #136-Ubuntu SMP Fri Jun 10 13:40:48 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

### Orchestration system version in use (e.g. `kubectl version`, ...)
```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.6+k3s1", GitCommit:"3228d9cb9a4727d48f60de4f1ab472f7c50df904", GitTreeState:"clean", BuildDate:"2022-01-25T01:27:44Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
```
### Link to relevant artifacts (policies, deployments scripts, ...)
```yaml
apiVersion: security.kubearmor.com/v1
ind: KubeArmorPolicy
etadata:
 name: ksp-wordpress-block-config
 namespace: wordpress-mysql
pec:
 severity: 10
 message: "blocked access to wordpress configuration file"
 selector:
   matchLabels:
     app: wordpress
 file:
   matchPaths:
   - path: /var/www/html/wp-config.php
     fromSource:
     - path: /bin/cat
 action:
   Block
```  

### Target containers/pods
`kubectl apply -f https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/wordpress-mysql/wordpress-mysql-deployment.yaml`

## To Reproduce

1. Install wordpress-mysql deployment `kubectl apply -f https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/wordpress-mysql/wordpress-mysql-deployment.yaml`
2. Apply policy specified above
3. Execute `kubectl exec -it -n wordpress-mysql wordpress-XXXXXXXX -- bash -c "cat wp-config.php"`
4. Execute `kubectl exec -it -n wordpress-mysql wordpress-XXXXXXXX -- bash -c "cat /var/www/html/wp-config.php"`

### Actual Behavior
1. Access to `wp-config.php` should be denied. (This is working)
2. The alert that is generated should have `PolicyName: ksp-wordpress-block-config`.
    * With Step 3 above, this PolicyName is `DefaultPosture` 
    * With Step 4 above, the policy name is appropriately set.

### Expected Behavior
The PolicyName should be correctly set to `ksp-wordpress-block-config` in both the cases where absolute or relative path accesses are made.

```diff
{
  "Timestamp": 1656682358,
  "UpdatedTime": "2022-07-01T13:32:38.159601Z",
  "ClusterName": "default",
  "HostName": "ubuntu2004-vagrants",
  "NamespaceName": "wordpress-mysql",
  "PodName": "wordpress-bf95888cb-jvcxh",
  "Labels": "app=wordpress",
  "ContainerID": "527562795ec2c4b68573e25f5cddb33c8114117a012c588a7121407e684fcf30",
  "ContainerName": "wordpress",
  "ContainerImage": "docker.io/library/wordpress:4.8-apache@sha256:6216f64ab88fc51d311e38c7f69ca3f9aaba621492b4f1fa93ddf63093768845",
  "HostPPID": 1897055,
  "HostPID": 1897061,
  "PPID": 1897055,
  "PID": 1330,
  "ProcessName": "/bin/cat",
- "PolicyName": "DefaultPosture",     # <----- Actual
+ "PolicyName": "ksp-wordpress-block-config",    # <---- Expected
  "Type": "MatchedPolicy",
  "Source": "/bin/cat wp-config.php",
  "Operation": "File",
  "Resource": "wp-config.php",
  "Data": "syscall=SYS_OPEN flags=O_RDONLY",
  "Enforcer": "AppArmor",
  "Action": "Block",
  "Result": "Permission denied"
}
```

### PR DESCRIPTION
**Pull request by Prateeknandle**
**Purpose of PR?**:

Fixes #750 

**Does this PR introduce a breaking change?**
no

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [x] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: KubeArmor/feeder/policyMatcher.go [hunk 1]
```diff
@@ -1080,6 +1080,10 @@ func (fd *Feeder) UpdateMatchedPolicy(log tp.Log) tp.Log {
 						matchedRegex = strings.HasSuffix(log.ProcessName, "/"+secPolicy.Resource) // processpath = */execname
 					}
 
+					// fallback logic: in case we get relative path in log.Resource then we join cwd + resource to get pull path
+					if !strings.HasPrefix(strings.Split(log.Resource, " ")[0], "/") && log.Cwd != "/" {
```
Comment (daemon1024) [inline KubeArmor/feeder/policyMatcher.go]: Do you think there will be a case where, 
Cwd[0] != "/"?
That is we have a relative path in CWD.

File: KubeArmor/feeder/policyMatcher.go [hunk 2]
```diff
@@ -936,7 +936,7 @@ func matchResources(secPolicy tp.MatchPolicy, log tp.Log) bool {
 		}
 		if secPolicy.ResourceType == "Directory" && (strings.HasPrefix(firstLogResourceDir, secPolicy.Resource) &&
 			((!secPolicy.Recursive && firstLogResourceDirCount == strings.Count(secPolicy.Resource, "/")) ||
-				(secPolicy.Recursive && firstLogResourceDirCount >= strings.Count(secPolicy.Resource, "/")))) || (secPolicy.Resource == (log.Resource + "/")) {
+				(secPolicy.Recursive && firstLogResourceDirCount >= strings.Count(secPolicy.Resource, "/")))) || (secPolicy.Resource == (log.Resource + "/")) || (secPolicy.Resource == "/" && secPolicy.Recursive) {
```
Comment (daemon1024) [inline KubeArmor/feeder/policyMatcher.go]: Please add a comment here about all the things we are checking.

File: KubeArmor/feeder/policyMatcher.go (hunk 1 above)
Comment (Prateeknandle) [reply KubeArmor/feeder/policyMatcher.go]: I don't think we can get relative path in cwd

**Review by daemon1024 — DISMISSED** [review]

**Comment by Prateeknandle** [comment]
now passing the generated full path to `BuildPidNode()`, in order to get updated processName as well.

**Review by Aryan-sharma11 — APPROVED** [review]
LGTM

**Review by daemon1024 — APPROVED** [review]

**Pull request merged**